### DISCUSSION TITLE
How does HeaterGauge work exactly?

### DISCUSSION THREAD

**Discussion by KlausHans** [julianschill/klipper-led_effect#210] [Q&A] [answered]
Thank you for this great piece of software.

However i am struggling to get HeaterGauge to work (other effects i tried work fine). This is my config:
```
[neopixel case]
# SK6812
pin: PA8
chain_count: 15
color_order: GRBW
initial_RED: 0.0
initial_GREEN: 0.0
initial_BLUE: 0.0
initial_WHITE: 0.1

[led_effect heating]
autostart:              false
frame_rate:             24
leds:
    neopixel:case
layers:
    heatergauge  15 0 top (0.0,0.0,1.0,0.0),(1.0,0.0,0.0,0.0)
heater: extruder
```

If i set a temperature of the extruder via fluid (e.g. `SET_HEATER_TEMPERATURE HEATER=extruder TARGET=150`) nothing happens, the case leds are still white from the init.

Is this bugged or is the error on my side?
What is the meaning of trailing and leading leds exactly?
Thanks!

**Comment by julianschill** [comment, marked as answer]
Did you activate the effect with `SET_LED_EFFECT EFFECT=heating` before setting the temperature?

**Reply by KlausHans** [reply]
Thank you, that was the problem, i didn't. I assumed somehow the effect is automatically set when heating.
